src/Curl/Curl.php

```diff
@@ -199,14 +199,12 @@ public function addResponseHeaderLine($curl, $header_line)
         return strlen($header_line);
     }
 
-    // protected methods
-
     /**
      * Execute the curl request based on the respective settings.
      *
      * @return int Returns the error code for the current curl request
      */
-    protected function exec()
+    public function exec()
```

Comment:
I am not against setting up this function public. but its return is not suitable for that.

Maybe we can rename the internal method and add a new `exec` method or simply adding new `execute` method.